contracts/factories/DelegatedManagerFactory.sol

```diff
@@ -0,0 +1,372 @@
+     *
+     * @return (ISetToken, address) The created SetToken and DelegatedManager addresses, respectively
+     */
+    function createSetAndManager(
+        address[] memory _components,
+        int256[] memory _units,
+        string memory _name,
+        string memory _symbol,
+        address _owner,
+        address _methodologist,
+        address[] memory _modules,
+        address[] memory _operators,
+        address[] memory _assets,
+        address[] memory _extensions
+    )
+        external
+        returns (ISetToken, address)
+    {
+        require(_extensions.length > 0, "Must have at least 1 extension");
+
+        if (_assets.length != 0) {
+            _validateComponentsIncludedInAssetsList(_components, _assets);
+        }
+
+        ISetToken setToken = _deploySet(
+            _components,
+            _units,
+            _modules,
+            _name,
+            _symbol
+        );
+
+        DelegatedManager manager = _deployManager(
+            setToken,
+            _methodologist,
+            _extensions,
+            _operators,
+            _assets
+        );
+
+        _setInitializationState(setToken, address(manager), _owner);
+
+        return (setToken, address(manager));
+    }
+
+    /**
+     * ONLY SETTOKEN MANAGER: Deploys a DelegatedManager and sets some temporary metadata about the
+     * deployment which will be read during a subsequent intialization step which wires everything together.
+     * This method is used when migrating an existing SetToken to the DelegatedManager system.
+     *
+     * (Note: This flow should work well for SetTokens managed by an EOA. However, existing
+     * contract-managed Sets may need to have their ownership temporarily transferred to an EOA when
+     * migrating. We don't anticipate high demand for this migration case though.)
+     *
+     * @param  _setToken         Instance of SetToken to migrate to the DelegatedManager system
+     * @param  _owner            Address to set as the DelegateManager's `owner` role
+     * @param  _methodologist    Address to set as the DelegateManager's methodologist role
+     * @param  _operators        List of operators authorized for the DelegateManager
+     * @param  _assets           List of assets DelegateManager can trade. When empty, asset allow list is not enforced
+     * @param  _extensions       List of extensions authorized for the DelegateManager
+     *
+     * @return (address) Address of the created DelegatedManager
+     */
+    function createManager(
+        ISetToken _setToken,
+        address _owner,
+        address _methodologist,
+        address[] memory _operators,
+        address[] memory _assets,
+        address[] memory _extensions
+    )
+        external
+        returns (address)
+    {
+        require(_extensions.length > 0, "Must have at least 1 extension");
+
+        if (_assets.length != 0) {
```

Comment:
Small nit: Can we just put the _extension check and this _assets check into a `_validateManagerParameters` function